src/services/CleanupService.ts

```diff
@@ -952,7 +952,7 @@ export const cleanCookiesOperation = async (
     // Don't store domains for private browsing data
     if (storesIdsToScrub.includes(id)) continue;
     for (const sd of SITEDATATYPES) {
-      if ((storeResults[sd] || []).length > 0) {
+      if (storeResults && (storeResults[sd] || []).length > 0) {
```

Comment:
While this may be less code to write, I think the check should be earlier on so as not to keep checking again?

i.e. just below line 953 (before the for loop) we could put
```javascript
if (!storeResults) continue;
```
so that the for loop is completely bypassed if that object doesn't exist.

Reply:
Done. As a side note, I didn't realize that that line was in a for loop until I scrolled way up. Perhaps it might time to refactor that large function in the future?

Reply:
I'm not sure I see much to refactor considering the inner workings aren't used anywhere else.  There's the isSafeToClean object generation (already separate), the debug sanitization steps (relies on debug pref and isSafeToCleanObject), the cookieStoreId generations (relies on cache browser), then the for loop for each container, to which it gets all cookies for each container and filters it through isSafeToClean object before cleaning cookies, and similar story in otherBrowsingDataCleanup but the entire isSafeToCleanObject is passed in for further filtering.